### PR TITLE
Fapi backport fixes part3

### PR DESCRIPTION
**Pull request by JuergenReppSIT**
* The usage of hierarchies in provisioning and Fapi_Delete was re-factored.
* Several segfault problems were fixed.
* Event logs were checked before Fapi_Extend.
* To find segfaults asserts which can be enabled with --enable-debug were added to
  the integration tests.

Signed-off-by: Christian Plappert <christian.plappert@sit.fraunhofer.de>
Signed-off-by: Juergen Repp <juergen.repp@sit.fraunhofer.de>

### CODE REVIEW

File: src/tss2-fapi/api/Fapi_PcrRead.c [hunk 1]
```diff
@@ -266,6 +274,8 @@ Fapi_PcrRead_Finish(
 
 cleanup:
     /* Cleanup any intermediate results and state stored in the context. */
+    if (r)
+        SAFE_FREE(command->pcrValue);
     SAFE_FREE(command->pcrValues);
```
Comment (tstruk) [inline src/tss2-fapi/api/Fapi_PcrRead.c:279]: Either this if block or the line after not needed

Comment (JuergenReppSIT) [reply src/tss2-fapi/api/Fapi_PcrRead.c:279]: Two different variables are freed. I will add a comment.

File: src/tss2-fapi/fapi_util.c [hunk 2]
```diff
@@ -4025,7 +4025,22 @@ ifapi_get_certificates(
...
+        return_if_error2(r, "Could not open hierarchy /HS");
+
+        fallthrough;
+
+    statecase(context->get_cert_state, GET_CERT_GET_CERT_READ_HIERARCHY);
+        r = ifapi_keystore_load_finish(&context->keystore, &context->io,
+                                       &context->nv_cmd.auth_object);
+        return_try_again(r);
+        return_if_error(r, "read_finish failed");
+
```
[the remark this answers is not in the document]
Comment (JuergenReppSIT) [reply src/tss2-fapi/fapi_util.c:4111]: I will change it for all checks of finish calls in this function.

File: src/tss2-fapi/api/Fapi_PcrRead.c (hunk 1 above)
Comment (tstruk) [reply src/tss2-fapi/api/Fapi_PcrRead.c:281]: Yes, sorry, I'm blind

**Pull request merged**